codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2FrameWriter.java

```diff
@@ -102,6 +105,7 @@ public void close() { }
     @Override
     public ChannelFuture writeData(ChannelHandlerContext ctx, int streamId, ByteBuf data,
             int padding, boolean endStream, ChannelPromise promise) {
+        SimpleChannelPromiseAggregator promiseAggregator = null;
```

Comment:
Assuming this class stays local to the http/2 package we may be OK (null checks may not be necessary for current uses) with the exceptions.  However the usage pattern is still a bit different given the two interfaces.

Lets say for example the class changes to the following:
```java
SimpleChannelPromiseAggregator(ChannelHandlerContext ctx, ChannelPromise promise) {
...
}

public ChannelPromise newPromise() {
  ++expectedCount;
 ...
}
```

If someone then creates this and uses it as follows:
```java
SimpleChannelPromiseAggregator ag = new SimpleChannelPromiseAggregator(ctx, promise);
someSynchronousSuccessMethod(ag.newPromise());
// Promise will be completed here and future calls will have no impact on promise
```

If `someSynchronousSuccessMethod` synchronously completes its promise argument then `promise` will be completed because `SimpleChannelPromiseAggregator` expects 1 promise to complete.

So the user has to know to pre-allocate all promises they intend to use (if `SimpleChannelPromiseAggregator` isn't told the total amount up front).